app/controllers/practices_controller.rb

```diff
@@ -537,9 +570,36 @@ def set_practice
     @practice = Practice.friendly.find(id)
   end
 
+  def practice_locked_for_editing
+    cur_user_id = current_user[:id]
+    locked_rec = PracticeEditorSession.practice_locked(@practice.id)
+    # if not locked - lock the practice for editing (for the current user)
+    if locked_rec == 0
+      PracticeEditorSession.lock_practice_for_user(cur_user_id, @practice.id)
+    else
+      if PracticeEditorSession.session_out_of_time(locked_rec)
+        PracticeEditorSession.lock_practice_for_user(cur_user_id, @practice.id)
+        return
+      end
+      locked_by_user_id = PracticeEditorSession.locked_by_user(locked_rec)
+      if locked_by_user_id != cur_user_id
+        locked_by = PracticeEditorSession.locked_by(locked_rec, false)
+        is_admin = PracticeEditorSession.is_admin(cur_user_id)
```

Comment:
we don't need to create this separate method is_admin to check the role... I believe we can just call user_role on the User model instance. And we should be getting the user_role of whoever locked it